pkg/scheduler/queue/queue.go

```diff
@@ -140,12 +142,12 @@ FindQueue:
 
 		// Pick next request from the queue.
 		for {
-			request := <-queue
-			if len(queue) == 0 {
+			request := <-queue.Chan()
+			if len(queue.Chan()) == 0 {
```

Comment:
I think this needs to use the new `queue.Len` since the new queue impl is nested and `len(ch)` won't test the recursive sub queues.

edit: you've already addressed in https://github.com/grafana/loki/pull/8722